components/plugin_marketplace/marketplace_item/marketplace_item.js

```diff
@@ -4,16 +4,197 @@
+                        id='marketplace_modal.list.update'
+                        defaultMessage='Update'
+                    />
+                </a>
+            </b>
+        </div>
+    );
+};
+
+UpdateDetails.propTypes = {
+    version: PropTypes.string.isRequired,
+    releaseNotesUrl: PropTypes.string,
+    installedVersion: PropTypes.string,
+    isInstalling: PropTypes.bool.isRequired,
+    onUpdate: PropTypes.func.isRequired,
+};
+
+// UpdateConfirmationModal prompts before allowing upgrade, specially handling major version changes.
+export const UpdateConfirmationModal = ({show, name, version, installedVersion, releaseNotesUrl, onUpdate, onCancel}) => {
+    if (!installedVersion) {
+        return null;
+    }
+
+    var isUpdate = false;
+    try {
+        isUpdate = semver.gt(version, installedVersion);
+    } catch (e) {
+        // If we fail to parse the version, assume not an update;
+    }
+
+    if (!isUpdate) {
+        return null;
+    }
+
+    const messages = [(
+        <FormattedHTMLMessage
+            id='marketplace_modal.list.update_confirmation.message.intro'
+            key='intro'
+            defaultMessage={`<p>Are you sure you want to update the ${name} plugin to ${version}?</p>`}
+            values={{name, version}}
+        />
+    )];
+
+    if (releaseNotesUrl) {
+        messages.push(
+            <FormattedHTMLMessage
+                id='marketplace_modal.list.update_confirmation.message.current_with_release_notes'
+                key='current'
+                defaultMessage={`<p>You currently have ${installedVersion}. View the <a href='${releaseNotesUrl}' target='_blank' rel='noopener noreferrer'>Release Notes</a> to learn about the changes included in this update.</p>`}
+                values={{installedVersion, releaseNotesUrl}}
+            />
+        );
+    } else {
+        messages.push(
+            <FormattedHTMLMessage
+                id='marketplace_modal.list.update_confirmation.message.current'
+                key='current'
+                defaultMessage={`<p>You currently have ${installedVersion}.</p>`}
+                values={{installedVersion}}
+            />
+        );
+    }
+
+    let sameMajorVersion = false;
+    try {
+        sameMajorVersion = semver.major(version) === semver.major(installedVersion);
+    } catch (e) {
+        // If we fail to parse the version, assume a potentially breaking change.
+        // In practice, this won't happen since we already tried to parse the version above.
+    }
+
+    if (!sameMajorVersion) {
+        if (releaseNotesUrl) {
+            messages.push(
+                <FormattedHTMLMessage
+                    id='marketplace_modal.list.update_confirmation.message.warning_major_version'
+                    key='warning'
+                    defaultMessage={'<p class=\'alert alert-warning\'>This update may contain breaking changes. Consult the release notes above before upgrading.</p>'}
```

Comment:
Why not extract the `<p>` tag so the message does not include the class names information? 

In the future when we want to change the class names here we would have to go to all translated files to update the message. Maybe something like: 

```
<p class=\'alert alert-warning\'>
	<FormattedHTMLMessage
	                    id='marketplace_modal.list.update_confirmation.message.warning_major_version'
	                    key='warning'
	                    defaultMessage={'This update may contain breaking changes. Consult the release notes above before upgrading.'}
	/>
</p>
```

Reply:
Ah, this is so much better. Thank you!!